### DISCUSSION TITLE
Time considered at home

### DISCUSSION THREAD

**Discussion by davide1295** [Vaskivskyi/ha-asusrouter#210] [General]
Hi, congratulations on the job.  I have a doubt / request about the operation of the tracker device.  Does the time in which a device is considered in the house depend on the update time of the sensors?  For example if I set a scan interval of 1 minute the device will not be detected in the house until it is scanned every minute right? 
It would be interesting to set a different time for updating sensors and device trackers

**Comment by Vaskivskyi** [comment]
Hello, @davide1295,

The simple answer is yes, in your example device will not change status before this one minute finishes.

So how this works currently is in the following way. The `scan interval` which you select during the configuration is the interval at which all the sensors are polled, including `device_tracker`s. The `consider home` value is an interval after which the device will be marked as `not home`.

So at this point, you have 2 options:
1. If `consider home` (e.g. 5 seconds) is smaller than `scan interval` (e.g. 10 seconds), as soon as sensors are polled and the device is not available - it will be marked `not home`. As soon as 10 seconds pass, the device status will be updated from `home` to `not home`.
2. If `consider home` (e.g. 15 seconds) is larger than `scan interval` (e.g. 10 seconds), the first sensor polling will notice the device being offline, but will still keep it `home`. Only the second update will change the state to `not home`.

Please, notice, that the change from `not home` to `home` will happen immediately on sensors polling regardless of the above rules.

Answering the last part, there are already plans to split device trackers polling from all other sensors polling. This feature will be implemented in one of the next versions of AsusRouter
